haiku/_src/stateful.py

```diff
@@ -122,12 +122,8 @@ def grad(fun, argnums=0, has_aux=False, holomorphic=False):
                                       holomorphic=holomorphic)
 
   def grad_fn(*args, **kwargs):
-    value, grads = value_and_grad_fun(*args, **kwargs)
-    if has_aux:
-      value, aux = value
-      return grads, aux
-    else:
-      return grads
+    (_, aux), grads = value_and_grad_fun(*args, **kwargs)
+    return (grads, aux if hax_aux else grads)
```

Comment:
Won't this return `(grads, grads)` when `hax_aux=False`? That would be different from what it did before.

Reply:
You're right. The problem has been resolved and reflected. 44cc50